test/common/upstream/ring_hash_lb_test.cc

```diff
@@ -523,6 +537,8 @@ TEST_P(RingHashLoadBalancerTest, HostAndLocalityWeightedTinyRing) {
   }
 }
 
+// Given both host weights and locality weights, and a sufficiently large ring, expect that requests
+// will distribute to the hosts with approximately the right proportion.
```

Comment:
Can we add one test that does the boundary cases of weight extremities, e.g. with localities of 1, host weights of 1, and localities of 127, host weights of 127; unless this is already covered somewhere?

Reply:
Ok, I've added another test case to cover that more explicitly.